packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js

```diff
@@ -19,33 +23,69 @@ export const propTypes = {
   /** Callback called when application launcher toggle is clicked */
   onToggle: PropTypes.func,
   /** Indicates where menu will be alligned horizontally */
-  position: PropTypes.oneOf(Object.values(DropdownPosition)),
+  position: PropTypes.oneOf(Object.values(ApplicationLauncherPosition)),
   /** Adds accessible text to the button. Required for plain buttons */
   'aria-label': PropTypes.string
 };
 
 export const defaultProps = {
   className: '',
-  direction: DropdownDirection.down,
+  direction: ApplicationLauncherDirection.down,
   dropdownItems: [],
   isOpen: false,
   onSelect: Function.prototype,
   onToggle: Function.prototype,
-  position: DropdownPosition.left,
-  'aria-label': defaultAriaLabel
+  position: ApplicationLauncherPosition.left,
+  'aria-label': 'Actions'
 };
 
-const ApplicationLauncher = ({ 'aria-label': ariaLabel, onToggle, ...props }) => (
-  <Dropdown
-    {...props}
-    toggle={
-      <DropdownToggle aria-label={ariaLabel} iconComponent={null} onToggle={onToggle}>
-        <ThIcon />
-      </DropdownToggle>
-    }
-    isPlain
-  />
-);
+class ApplicationLauncher extends React.Component {
+  onEnter = () => {
+    this.openedOnEnter = true;
+  };
+
+  componentDidUpdate() {
+    if (!this.props.isOpen) this.openedOnEnter = false;
+  }
+
+  render() {
+    const {'aria-label': ariaLabel, children, dropdownItems, className, isOpen, onSelect, onToggle, ...props} = this.props;
+    return <GenerateId>{randomId => (
+      <div
+        className={css(
+          styles.appLauncher,
+          isOpen && styles.modifiers.expanded,
+          className
+        )}
+        ref={ref => {
+            this.parentRef = ref;}}>
+        {Children.map(
+          <Toggle id={`pf-toggle-id-${randomId}`} aria-label={ariaLabel} onToggle={onToggle}><ThIcon /></Toggle>, oneToggle =>
+            cloneElement(oneToggle, {
+              parentRef: this.parentRef,
+              id: randomId,
```

Comment:
should id here match line 63?

Reply:
I think so. That's how it was originally.